apps/dolly-frontend/src/main/js/src/components/fagsystem/pdlf/form/partials/familierelasjoner/foreldreansvar/Foreldreansvar.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react'
+import { FormikProps } from 'formik'
+import { Option, SelectOptionsOppslag } from '~/service/SelectOptionsOppslag'
+import {
+	initialForeldreansvar,
+	initialPdlBiPerson,
+	initialPdlPerson,
+} from '~/components/fagsystem/pdlf/form/initialValues'
+import { AvansertForm } from '~/components/fagsystem/pdlf/form/partials/avansert/AvansertForm'
+import { FormikDollyFieldArray } from '~/components/ui/form/fieldArray/DollyFieldArray'
+import { FormikSelect } from '~/components/ui/form/inputs/select/Select'
+import { SelectOptionsManager as Options } from '~/service/SelectOptions'
+import { FormikDatepicker } from '~/components/ui/form/inputs/datepicker/Datepicker'
+import _get from 'lodash/get'
+import { AdresseKodeverk } from '~/config/kodeverk'
+import { PdlPersonForm } from '~/components/fagsystem/pdlf/form/partials/pdlPerson/PdlPersonForm'
+import _cloneDeep from 'lodash/cloneDeep'
+import _set from 'lodash/set'
+import { AlertStripeAdvarsel } from 'nav-frontend-alertstriper'
+import Loading from '~/components/ui/loading/Loading'
+import { ForeldreBarnRelasjon } from '~/components/fagsystem/pdlf/PdlTypes'
+
+interface ForeldreansvarForm {
+	formikBag: FormikProps<{}>
+	identOptions: Array<Option>
+	loadingOptions: boolean
+}
+
+type Target = {
+	label: string
+	value: string
+}
+
+export const Foreldreansvar = ({ formikBag, identOptions, loadingOptions }: ForeldreansvarForm) => {
+	const navnInfo = SelectOptionsOppslag.hentPersonnavn()
+	//@ts-ignore
+	const fornavnOptions = SelectOptionsOppslag.formatOptions('fornavn', navnInfo)
+	//@ts-ignore
+	const mellomnavnOptions = SelectOptionsOppslag.formatOptions('mellomnavn', navnInfo)
+	//@ts-ignore
+	const etternavnOptions = SelectOptionsOppslag.formatOptions('etternavn', navnInfo)
+
+	const handleChangeTypeAnsvarlig = (target: Target, path: string) => {
+		const foreldreansvar = _get(formikBag.values, path)
+		const foreldreansvarClone = _cloneDeep(foreldreansvar)
+
+		_set(foreldreansvarClone, 'typeAnsvarlig', target?.value || null)
+		if (!target) {
+			_set(foreldreansvarClone, 'ansvarlig', undefined)
+			_set(foreldreansvarClone, 'ansvarligUtenIdentifikator', undefined)
+			_set(foreldreansvarClone, 'nyAnsvarlig', undefined)
```

Comment:
Kanskje sette stringene "ansvarlig", "ansvarligUtenIdentifikator", og "nyAnsvarlig" som egne consts for å unngå potensielle skrivefeil siden dem blir brukt veldig mye.

Reply:
Veldig lurt 😄